pppd/tty.c

```diff
@@ -559,7 +559,7 @@ int connect_tty(void)
 				status = EXIT_OPEN_FAILED;
 				goto errret;
 			}
-			real_ttyfd = open(devnam, O_NONBLOCK | O_RDWR, 0);
+			real_ttyfd = open(devnam, O_NONBLOCK | O_RDWR | O_CLOEXEC, 0);
```

Comment:
This will prevent the tty file descriptor from being accessed by the connector script.